app/assets/javascripts/discourse/app/components/post/menu/buttons/like.gjs

```diff
@@ -82,17 +92,112 @@ export default class PostMenuLikeButton extends Component {
       </div>
     {{else}}
       <div class="double-button">
-        <LikeCount
-          ...attributes
-          @action={{@buttonActions.toggleWhoLiked}}
-          @state={{@state}}
-          @post={{@post}}
-        />
+        {{#if @post.likeCount}}
+          <LikedUsersList ...attributes @post={{@post}} />
+        {{else}}
+          <LikeCount
+            ...attributes
+            @action={{@buttonActions.toggleWhoLiked}}
+            @state={{@state}}
+            @post={{@post}}
+          />
+        {{/if}}
       </div>
     {{/if}}
   </template>
 }
 
+class LikedUsersList extends Component {
+  @service store;
+
+  @tracked likedUsers = null;
```

Comment:
the smallest of nits - unless you specifically need `null`, it's cool to leave it uninitialized:

```suggestion
  @tracked likedUsers;
```